### PR TITLE
upgrade go kubernetes packages to v0.23.8

### PR DESCRIPTION
**Pull request by i332322**
upgrading go kubernetes packges from v0.23.5 to v.023.8
due to security vulerability issues raised @Whitesource:

/kubernetes-v1.23.5:CVE-2022-1996 
/kubernetes-v1.23.5:CVE-2022-27191 
/kubernetes-v1.23.5:CVE-2022-21698
/kubernetes-v1.23.5:CVE-2021-43565

> go mod tidy && make build-local
> env CGO_ENABLED=0 go build -o kubeconfig-service ./cmd/generator/main.go 
> SUCCESS

### CODE REVIEW

File: components/kubeconfig-service/go.mod [hunk 1]
```diff
@@ -71,31 +71,33 @@ require (
 )
 
 replace (
-	k8s.io/api => k8s.io/api v0.23.5
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.23.5
+	github.com/emicklei/go-restful => github.com/emicklei/go-restful/v3 v3.7.5-0.20220308211933-7c971ca4d0fd
+	golang.org/x/crypto => golang.org/x/crypto v0.0.0-20220314234659-1baeb1ce4c0b
+	k8s.io/api => k8s.io/api v0.23.8
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.23.8
 	k8s.io/apimachinery => k8s.io/apimachinery v0.23.6-rc.0
```
Comment (ebensom) [inline components/kubeconfig-service/go.mod:78]: I would also bump all these v0.23.6-rc.0 versions to v0.23.8

File: components/kubeconfig-service/go.mod [hunk 2]
```diff
@@ -71,31 +71,33 @@ require (
 )
 
 replace (
-	k8s.io/api => k8s.io/api v0.23.5
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.23.5
+	github.com/emicklei/go-restful => github.com/emicklei/go-restful/v3 v3.7.5-0.20220308211933-7c971ca4d0fd
```
Comment (ebensom) [inline components/kubeconfig-service/go.mod:74]:
```suggestion
	github.com/emicklei/go-restful => github.com/emicklei/go-restful/v3 v3.8.0
```

File: components/kubeconfig-service/go.mod [hunk 3]
```diff
@@ -71,31 +71,33 @@ require (
...
+	golang.org/x/crypto => golang.org/x/crypto v0.0.0-20220314234659-1baeb1ce4c0b
```
Comment (ebensom) [inline components/kubeconfig-service/go.mod:75]: I think we noticed that there are two CVEs, and one of them is only fixed in a March 15 version, which is referenced by the github.com/kubernetes/kubernetes master branch

File: components/kubeconfig-service/go.mod (hunk 2 above)
Comment (i332322) [reply components/kubeconfig-service/go.mod]: I've corrected as suggested.

File: components/kubeconfig-service/go.mod (hunk 3 above)
Comment (i332322) [reply components/kubeconfig-service/go.mod:75]: yes you're right. emicklei/go-restful package i've fixed and using the later version v3.8.0 which the v.1.25.0-alpha version of kubernetes use. We have seen that make build-local has successfully ran.

**Review by ebensom — DISMISSED** [review]

**Pull request closed without merging**